### PR TITLE
feat(mep): Extract custom measurements [INGEST-1389 INGEST-1417]

### PR DESCRIPTION
**Pull request by jjbayer**
Up until now, metrics are only extracted from transaction payloads if they are in the `extractMetrics` allowlist passed down from sentry through project config.

In order to allow custom measurements, we add another field to the project config which defines how many of them we accept, for example: `{ "customMeasurements": { "limit": 5 } }`.

### CODE REVIEW

**Review by untitaker — COMMENTED** [review]
When talking with @jan-auer about this on monday I believe he preferred that we limit the total number of measurements, but I'm fine with eitehr way

**Comment by jjbayer** [comment]
> When talking with @jan-auer about this on monday I believe he preferred that we limit the total number of measurements, but I'm fine with eitehr way

I believe limiting the number of custom measurements is easier to maintain. When we add a new builtin measurement, we don't want to accidentally decrease the user's budget for custom measurements.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 **Features**:
 
 - Extract metrics also from trace-sampled transactions. ([#1317](https://github.com/getsentry/relay/pull/1317))
+- Extract metrics from a configurable amount of custom transaction measurements. ([#1324](https://github.com/getsentry/relay/pull/1324))
```
Comment (jan-auer) [inline CHANGELOG.md]: This could probably go into "Internal"

**Review by jan-auer — APPROVED** [review]
From offline conversations, we've circled in on limiting just the number of custom measurements to make the product experience more robust against SDK changes. We can still enforce another overall limit if Snuba needs it.

The nesting level for the measurements limit seems a bit superfluous, especially since `extract_metrics` also relates to custom measurements. However, if you expect more similar options, this is good to keep.

**Review by flub — APPROVED** [review]

**Pull request merged**